lib/device-log/safari-network-log.js

```diff
@@ -166,6 +173,28 @@ class SafariNetworkLog extends RotatingLog {
       this.log.debug(`  Cancelled: ${cancelled}`);
     }
   }
+
+  async getLogs () {
+    if (this.logs.length && this.logIdxSinceLastRequest < this.logs.length) {
+      let result = _.clone(this.logs);
+      if (this.logIdxSinceLastRequest > 0) {
+        result = result.slice(this.logIdxSinceLastRequest);
+      }
+      this.logIdxSinceLastRequest = this.logs.length;
+
+      // in order to satisfy certain clients, we need to have a basic structure
+      // to the results, with `level`, `timestamp`, and `message`, which is
+      // all the information stringified
+      return result.map((entry) => {
+        return Object.assign({}, entry, {
```

Comment:
do we need `return` here? a shortcut `(entry) => Object.assign(...) might be used though